index.html

```diff
@@ -11604,6 +11604,7 @@ <h2>Definitions of States and Properties (all aria-* attributes)</h2>
 				<p><a>Indicates</a>, when set to <code>true</code>, that an <a>element</a> and its entire subtree are hidden from assistive technology, regardless of whether it is visibly rendered.</p>
 				<p>User agents determine an element's [=element/hidden=] status based on whether it is rendered, and the rendering is usually controlled by CSS. For example, an element whose <code>display</code> property is set to <code>none</code> is not rendered. An element is considered [=element/hidden=] if it, or any of its ancestors are not rendered or have their <code>aria-hidden</code> attribute value set to <code>true</code>.</p>
 				<p>Authors MAY, with caution, use aria-hidden to hide visibly rendered content from assistive technologies <em>only</em> if the act of hiding this content is intended to improve the experience for users of assistive technologies by removing redundant or extraneous content. Authors using aria-hidden to hide visible content MUST ensure that identical or equivalent meaning and functionality is exposed to assistive technologies.</p>
+				<p>Additionally, authors SHOULD ensure that any elements that are accessibility descendants of an aria-hidden element, or any elements which have been marked as aria-hidden themselves, are prevented from receiving focus. If an aria-hidden element or a descendant of an aria-hidden ancestor receives focus, User Agents MUST ignore the aria-hidden state of any ancestor to the focused element, resulting in the entire subtree to be exposed to assistive technologies from that point forward, even once the element loses focus or is no longer focusable.</p>
```

Comment:
```suggestion
				<p>Additionally, authors SHOULD prevent a focusable element from receiving focus if it or any of its accessibility ancestors have been <a href="#tree_exclusion">excluded from the accessibility tree</a>. If a focusable element receives focus while [=element/hidden=], User Agents MUST re-expose it and each of its [=element/hidden=] ancestors to the accessibility tree, and persist that exposure even after the triggering element loses focus or is no longer focusable.</p>
```

For the first sentence, here’s an editorial suggestion that reads a little more clearly to me:

> Additionally, authors SHOULD prevent a focusable element from receiving focus if it or any of its accessibility ancestors have been [excluded from the accessibility tree](https://www.w3.org/TR/wai-aria-1.3/#tree_exclusion).

In the second sentence, “_entire subtree_” sounds greedy. In this contrived example, if `#link` receives focus, I’m supposing we’d expect `#a` to get exposed (and therefore also `#b` and `#link`), but not `#c`:

```html
<div id="a" aria-hidden="true">
  <p id="b">Some stuff that is only hidden by virtue of its hidden parent.</p>
  <a id="link" href="#">I’m hidden but focusable</button>
  <p id="c" aria-hidden="true">A sibling to #link that is coincidentally but explicitly hidden.</p>
</div>
```

If that’s true, then how about this rephrasing?

> If a focusable element receives focus while [hidden](https://www.w3.org/TR/wai-aria-1.3/#dfn-hidden), User Agents MUST re-expose it and each of its [hidden](https://www.w3.org/TR/wai-aria-1.3/#dfn-hidden) ancestors to the accessibility tree, and persist that exposure even after the triggering element loses focus or is no longer focusable.

Reply:
We don't just expose the ancestor chain, but rather the entire subtree under the aria-hidden (basically the aria-hidden markup on the ancestor is marked as problematic and ignored by the browser, which can affect the entire subtree). Of course, there could be additional markup in some parts of the subtree that also hide things (display:none, etc.).

Reply:
i'm not a fan of switching between saying "excluded from the accessibility tree" and then switching to "hidden" (though i realize 'hidden' in this spec doesn't necessarily mean "completely hidden").

beyond what aaron mentioned (though maybe you updated the text since then) i think that's my only thing i'm not sure about with this rewording.  it uses the terminology correctly, but i had to read it a number of times to figure out if it still meant the same thing.  maybe that's just my problem though.